Automattic/wp-calypso
# Readymade Templates: Add Content with AI step implementation

Related to:
- https://github.com/Automattic/dotcom-forge/issues/7500

## Proposed Changes

* Implemented the `Generate content with AI` launchpad step.
* Added a `selectedReadymadeTemplate` to the onboarding store (action, reducer, etc)
* Created the Generate content step component.
* Created a `TextProgressBar` and a previewer component used by the step.

The new step is available on the Readymade Template flow and will generate content based on user inputs.
## Testing Instructions



* Open the Calypso live link
* Start the flow on `/patterns`
* Pick a Readymade Template 
* On the launchpad you should see a `Generate content with AI` step, click it
* Test the AI content generation screen.

Note: There are some rough edges and missing details that will be solved/added on a follow up.


https://github.com/user-attachments/assets/e980d6d2-721e-4644-9054-5f077213ca79



## Pre-merge Checklist
